File: backend/app/models/enums.py
# backend/app/models/enums.py
from enum import Enum

class TestamentType(str, Enum):
    OLD = "Old Testament"
    NEW = "New Testament"

class CanonicalType(str, Enum):
    ALL = "All"
    CATHOLIC = "Catholic"
    EASTERN_ORTHODOX = "Eastern Orthodox"
    PROTESTANT = "Protestant"
    NONE = "NONE"

class BookGroupType(str, Enum):
    TORAH = "Torah"
    HISTORICAL = "Historical"
    WISDOM = "Wisdom"
    MAJOR_PROPHETS = "Major Prophets"
    MINOR_PROPHETS = "Minor Prophets"
    GOSPELS = "Gospels"
    MODERN_HISTORICAL = "Modern Historical"
    PAULINE_EPISTLES = "Pauline Epistles"
    GENERAL_EPISTLES = "General Epistles"
    APOCALYPTIC = "Apocalyptic"

class BibleTranslation(str, Enum):
    KJV = "KJV"
    NIV = "NIV"
    ESV = "ESV"
    NASB = "NASB"
    NLT = "NLT"
    CSB = "CSB"
    NKJV = "NKJV"
    RSV = "RSV"
    MSG = "MSG"
    AMP = "AMP"

File: frontend/src/app/core/models/bible/enums.ts
// src/app/models/bible/enums.ts
// Enums for type safety

/**
 * Defines the Testament types in the Bible
 */
export enum TestamentType {
  OLD = 'Old Testament',
  NEW = 'New Testament'
}

/**
 * Defines the different book groups within the Bible
 */
export enum BookGroupType {
  LAW = 'Torah',
  HISTORY = 'Historical',
  WISDOM = 'Wisdom',
  MAJOR_PROPHETS = 'Major Prophets',
  MINOR_PROPHETS = 'Minor Prophets',
  GOSPELS = 'Gospels',
  MODERN_HISTORICAL = 'Modern Historical',
  PAULINE = 'Pauline Epistles',
  GENERAL = 'General Epistles',
  APOCALYPTIC = 'Apocalyptic'
}

/**
 * Constants for apocryphal chapters
 */
export const apocryphalChapters: Record<string, number[]> = {};
